Infrastructure/Specs/ReadMe.md

```diff
@@ -0,0 +1,228 @@
+    void sendMessageToContent(String message);    
+    void subscribeForNotifications(String identifier, contentNotificationHandler handler);
+}
+```
+```Java
+public interface contentNotificationHandler {
+    //Methods
+    void onMessageReceived(String username, String message, String image);            
+    void onNewUserJoined(String username);
+}
+
+```
+## Between Content Module and Networking module
+
+``` Java
+public interface INotificationHandler {
+    void onMessageReceived(String message);
+}
+```
+``` Java
+public interface ICommunicator {
+    #Start communication
+    void start();
+    #Stop the Communication
+    void stop();
+    #Send the data to the destination IP
+    void send(String destination, String message, String identifier);
+    #Subscribe the module for receiving the data from the network module
+    void subscribeForNotifications(String identifier, INotificationHandler handler);
+}
+```
+### Design Decisions
+* Content module interacts with UI module and Networking module parallel to the processing module
+* The content module uses the Icommunicator interface exposed by the Networking module to send data to the server
+* It uses the INotificationHandler interface exposed by the Networking to receive data from the server
+* A similar setup is used between content module and UI module for exchange of data
+* The messages are sent in plain text as no formatting requirement was given from UI side
+* The data is converted into a JSON format before sending to the server
+* The following fields was agreed for the JSON format as it gave complete information about the data being sent
+[Metadata]: Information on the type of data being sent(Image/Chat)
+[Client]: Information on the client/user sending the data(Username)
+[Data]: The actual data being sent
+* The message identifier being given as the value for the argument 'identifier' in the send() API given by networking, is given separately and also packaged into the JSON message so that the networking module can easily identify the type of message on the receiving client and send to the appropriate module without having to read the JSON message
+* Since the content module is handling sending and receiving on both interfaces, two queues are being used for the data flows 
+UI -> Content -> Networking and Networking -> Content -> UI
+* The content module after receiving the username, profile picture and server ip once from the UI at client initialisation will save these values in RAM and also notify the server of new username and profile so that the server can update its map of users and in turn notify other clients of the new client by sending its updated map of users to all clients which is locally stored by each client by their respective content module. This is done to have the profile images of all users available locally rather than send through the network when needed thereby reducing the load on the network
```

Comment:
~~will save these values in RAM~~ Doesn't sound good. It's obvious that things stay in memory unless mentioned for storage in the disk.

Reply:
Removed the sentence